src/resources/basis.js

```diff
@@ -0,0 +1,349 @@
+            return m.wasmUrl.indexOf(moduleName) !== -1;
+        });
+        if (!wasmModule) {
+            wasmModule = modules.find(function (m) {
+                return m.wasmUrl.indexOf('basist_all') !== -1;
+            });
+        }
+
+        if (pc.platform.workers) {
+            var code = '(' + BasisWorker.toString() + ')(' +
+                        'true,"' +
+                        this.urlBase + wasmModule.glueUrl + '","' +
+                        this.urlBase + wasmModule.wasmUrl + '","' +
+                        this.urlBase + wasmModule.fallbackUrl + '",' +
+                        '[' + formats.slice(0, 2).toString() + ']' +
+                        ')\n\n';
+            var blob = new Blob([code], { type: 'application/javascript' });
+            var url = URL.createObjectURL(blob);
+            this.worker = new Worker(url);
+            this.worker.addEventListener('message', this._handleWorkerResponse.bind(this));
+        } else {
+            this.transcoder = new BasisWorker(
+                false,
+                this.urlBase + wasmModule.glueUrl,
+                this.urlBase + wasmModule.wasmUrl,
+                this.urlBase + wasmModule.fallbackUrl,
+                formats.slice(0, 2));
+        }
+    };
+
+    // render thread worker manager
+    Basis.prototype = {
+        transcode: function (url, data, callback) {
+            if (this._isUrlRelative(url)) {
+                url = this.urlBase + url;
+            }
+            if (this.worker) {
+                if (!this.callbacks.hasOwnProperty(url)) {
+                    // store url and kick off worker job
+                    this.callbacks[url] = [callback];
+                    this.worker.postMessage({ url: url, data: data }, [data]);
+                } else {
+                    // the basis worker is already busy processing this url, store callback
+                    // (this shouldn't really happen since the asset system only requests
+                    // a resource once)
+                    this.callbacks[url].push(callback);
+                }
+            } else {
+                this.transcoder(url, data, function (err, data) {
+                    if (err) {
+                        callback(err, null);
+                    } else {
+                        data.format = this._basisToPixelFormat(data.format);
+                        callback(null, data);
+                    }
+                }.bind(this));
+            }
+        },
+
+        _isUrlRelative: function (url) {
+            return (url.indexOf('://')  <= 0 && url.indexOf('//') !== 0 );
+        },
+
+        _handleWorkerResponse: function (message) {
+            var url = message.data.url;
+            var err = message.data.err;
+            var data = message.data.data;
+            var callbacks = this.callbacks[url];
+
+            if (!callbacks) {
+                console.error('internal logical error encountered');
```

Comment:
Should we be a little more specific with those errors that they are coming from Basis?

Reply:
Sure, I'll indicate it's the basis transcoder